portal-kernel/src/com/liferay/document/library/kernel/store/StoreAreaAwareStoreWrapper.java

```diff
@@ -59,13 +60,13 @@ public void deleteDirectory(
 			StoreAreaProcessor storeAreaProcessor =
 				_storeAreaProcessorSupplier.get();
 
-			storeAreaProcessor.copyDirectory(
-				companyId, repositoryId, dirName, _SOURCE_STORE_AREAS,
-				StoreArea.DELETED);
+			if (storeAreaProcessor.copyDirectory(
+					companyId, repositoryId, dirName, _SOURCE_STORE_AREAS,
+					StoreArea.DELETED)) {
 
-			StoreArea.runWithStoreAreas(
-				() -> store.deleteDirectory(companyId, repositoryId, dirName),
-				StoreArea.LIVE, StoreArea.NEW);
+				_registerDeleteDirectoryCallback(
+					companyId, repositoryId, dirName, store);
+			}
```

Comment:
Some calls to DLStore are already invoked as a callback themselves in DLFileEntryLocalService. Can you double check that we're not re-registering callbacks unnecessary? If in doubt, assume DLStore will always be invoked after the transaction is finished (i.e. no need to register tx callbacks in DLStore).

Reply:
I checked and only the deleteFile method has one kind of registered callbacks on the service for the check-in and to delete the PWC. 
I could not found any conflict. with this callback and the service.